homeassistant/components/media_player/__init__.py

```diff
@@ -878,16 +893,21 @@ async def _async_fetch_image(hass, url):
                     content_type = response.headers.get(CONTENT_TYPE)
                     if content_type:
                         content_type = content_type.split(";")[0]
-                    cache_images[url][CACHE_CONTENT] = content, content_type
 
         except asyncio.TimeoutError:
             pass
 
-        while len(cache_images) > cache_maxsize:
-            cache_images.popitem(last=False)
-
         return content, content_type
 
+    def get_browse_image_url(self, browse_image):
+        """Generate an externally accessible url for a media browser image."""
+        base_url = get_url(self.hass, prefer_external=True)
```

Comment:
why prefer external?

Reply:
Also, we shouldn't do this. We should call the entities directly.

Reply:
Here was my intent: get an url that can be set as the thumbnail for the media browser code. The media browser expects an url, not the image itself. I set `prefer_external` so that images are viewable from outside the home network, for example, if using the android app.